roles/validate_inventory/tasks/ai.yml

```diff
@@ -28,3 +28,56 @@
     fail_msg: "{{ item }} is not within the machine network!"
   when: vip_dhcp_allocation == false
   loop: "{{ groups['masters'] + (groups['workers'] | default([])) }}"  # This should not include day2_workers as they can be RWNs
+
+- name: Validate extra VIPs for dualstack
+  when:
+    - extra_api_vip is defined
+    - extra_api_vip | length > 0
+    - extra_ingress_vip is defined
+    - extra_ingress_vip | length > 0
+  block:
+    - name: Assert that Openshift version is supported for dualstack VIPs
+      assert:
+        that:
+          - openshift_full_version is version('4.12', '>=')
+        fail_msg: "openshift_full_version  must be >= 4.12. to support dualstack VIPs"
+
+    - name: Check if extra api VIP is within the extra machine networks
+      ansible.builtin.debug:
+        msg: "{{ hostvars['assisted_installer']['extra_api_vip'] | ansible.utils.ipaddr(item.cidr) }}"
+      when:
+        - vip_dhcp_allocation == false
+        - extra_machine_networks is defined
+      register: extra_api_vip_tests
+      loop: "{{ hostvars['assisted_installer']['extra_machine_networks'] }}"
+
+    - name: Fail if extra api VIP is NOT within any of the extra machine networks
+      vars:
+        vip_field_filter: "results[*].msg"
+        extra_api_ipv6_list: "{{ extra_api_vip_tests | json_query(vip_field_filter) }}"
+      ansible.builtin.assert:
+        that:
+          - extra_api_vip in extra_api_ipv6_list
```

Comment:
Why does this have ipv6 in the name, is there an ordering requirement on the vips? Does it have to go ipv4 then ipv6? If so can you add a check for this ordering otherwise people might put them in the wrong order and not find out until lots of things have happened.

Reply:
not really, I was having in mind ipv6 for my tests all this time, so I renamed the variable to be independent of any ipv4 or ipv6